inc/saitypes.h

```diff
@@ -391,6 +391,64 @@ typedef struct _sai_ip_prefix_t
     sai_ip_addr_t mask;
 } sai_ip_prefix_t;
 
+typedef enum _sai_flex_data_type_t
+{
+    SAI_FLEX_DATA_TYPE_PRBS,
+
+    SAI_FLEX_DATA_TYPE_LAST
+
+} sai_flex_data_type_t;
+
+/**
+ * @brief Attribute data for #SAI_PORT_ATTR_PRBS_RX_STATUS
+ */
+typedef enum _sai_port_prbs_rx_status_t
+{
+    /** PRBS is locked and error_count is 0 */
+    SAI_PORT_PRBS_RX_STATUS_OK,
+
+    /** PRBS is locked, but there are errors */
+    SAI_PORT_PRBS_RX_STATUS_LOCK_WITH_ERRORS,
+
+    /** PRBS not locked */
+    SAI_PORT_PRBS_RX_STATUS_NOT_LOCKED,
+
+    /** PRBS locked but there is loss of lock since last call */
+    SAI_PORT_PRBS_RX_STATUS_LOST_LOCK,
+
+} sai_port_prbs_rx_status_t;
+
+/**
+ * @extraparam sai_flex_data_type_t type
+ */
+typedef union _sai_flex_status_t
+{
+    /** @validonly type == SAI_FLEX_DATA_TYPE_PRBS */
+    sai_port_prbs_rx_status_t rx_status;
+
+} sai_flex_status_t;
+
+/**
+ * @extraparam sai_flex_data_type_t type
+ */
+typedef union _sai_flex_data_t
```

Comment:
why this is union ? and contains only 1 element ? are you planning to extend that ?

Reply:
Yes, thats the idea. This is a generic data type for atomic I/O and more fields will be added later as needed